### ISSUE TITLE
Remember the loved songs on LastFM

### ISSUE THREAD

**Issue by moshpirit** [clementine-player/Clementine#4954]
When we mark as favourite a song in last.fm, the button is temporally selected, when we reboot Clementine, it's unselected (although marked as "loved" in the website), so the thing should be marked in the website _and_ in the music player in order to know what songs have we marked as favourites.

**Comment by xiongchiamiov** [comment]
This would be nice, although the problem comes about for things that have been loved outside the player (or in the past) - do you make an API call for each track to determine whether it's loved?

**Comment by moshpirit** [comment]
Good point, I don't know, but it's a step, so outside loved songs could come in the future. I'm not a developer so I have no idea of how to implement it :(

**Comment by dehare** [comment]
I've looked into the LastFM API and have to conclude that this has to be implemented by LastFM. When posting the current played track (i.e. track.updateNowPlaying), the API replies track details (name, artist, album) and it should therefore be relatively simple to implement a 'loved' boolean. From an end-point perspective, the API isn't adequate to find out if the current track is loved:
One would have to loop through all the loved tracks (user.getLovedTracks) and verify the info with the current song. Logically, if the response of getLovedTracks is long (i.e. many loved tracks), this task get very hardware intensive (noting the amount of requests per second).

See:
http://www.last.fm/api/show/track.updateNowPlaying
http://www.last.fm/api/show/user.getLovedTracks

**Comment by awsms** [comment]
[TauonMusicBox](https://github.com/Taiko2k/TauonMusicBox) has this functionnality. Would be great if it was on Clementine as well!
It would also be cool if we could generate a playlist from the tracks loved on Last.fm.
